pkg/component/etcd/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -0,0 +1,199 @@
+	policyv1 "k8s.io/api/policy/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type component struct {
+	client     client.Client
+	namespace  string
+	values     *Values
+	k8sVersion *semver.Version
+}
+
+// K8sVersionGetter defines a func type that accepts a rest configuration and returns the semantic k8s version of the k8s cluster
+type K8sVersionGetter func(config *rest.Config) (*semver.Version, error)
+
+const (
+	ownedByAnnotationKey   = "gardener.cloud/owned-by"
+	ownerTypeAnnotationKey = "gardener.cloud/owner-type"
+	instanceKey            = "instance"
+	nameKey                = "name"
+	appKey                 = "app"
+)
+
+// New creates a new poddisruptionbudget deployer instance.
+func New(c client.Client, namespace string, values *Values, k8sVersion semver.Version) gardenercomponent.Deployer {
+	return &component{
+		client:     c,
+		namespace:  namespace,
+		values:     values,
+		k8sVersion: &k8sVersion,
+	}
+}
+
+// getOwnerReferences returns owner references for the PDB object
+func getOwnerReferences(val *Values) []metav1.OwnerReference {
+	return []metav1.OwnerReference{
+		{
+			APIVersion:         druidv1alpha1.GroupVersion.String(),
+			Kind:               "Etcd",
+			Name:               val.EtcdName,
+			UID:                val.EtcdUID,
+			Controller:         pointer.BoolPtr(true),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+		},
+	}
+}
+
+// getLabels returns the labels for the PDB object
+func getLabels(val *Values) map[string]string {
+	labels := make(map[string]string)
+
+	for k, v := range val.Labels {
+		labels[k] = v
+	}
+
+	labels[instanceKey] = val.EtcdName
+	labels[nameKey] = "etcd"
+	labels[appKey] = "etcd-statefulset"
+
+	return labels
+}
+
+// getSelectorLabels returns the labels used in the selector of the PDB spec
+func getSelectorLabels(val *Values) map[string]string {
+	labels := map[string]string{
+		instanceKey: val.EtcdName,
+		nameKey:     "etcd",
+	}
+
+	return labels
+}
+
+// getAnnotations returns the annotations for the PDB object
+func getAnnotations(val *Values) map[string]string {
+	annotations := map[string]string{
+		ownedByAnnotationKey:   fmt.Sprintf("%s/%s", val.EtcdNameSpace, val.EtcdName),
+		ownerTypeAnnotationKey: "etcd",
+	}
+	return annotations
+}
+
+// emptyPodDisruptionBudget returns an empty PDB object with only the name and namespace as part of the object meta
+func (c *component) emptyPodDisruptionBudget(name string, k8sversion *semver.Version) (interface{}, error) {
+	k8sVersionGreaterEqual121, err := utils.CompareVersions(k8sversion.String(), ">=", "1.21")
+	if err != nil {
+		return nil, err
+	}
+	if k8sVersionGreaterEqual121 {
+		return &policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: c.namespace,
+			},
+		}, nil
+	} else {
```

Comment:
Add a TODO with your username to delete this else part when we won't be supporting k8s version lesser than 1.21. WDYT?

Reply:
Makes sense
Added comment